### DISCUSSION TITLE
How to handle database connections with Hot Reloading in API section

### DISCUSSION THREAD

**Discussion by fultonm** [vercel/next.js#23867] [Help] [answered]
My application creates a new database connection pool with `node-mysql2`. When calling a different API end points a new pool is created. Over time the RDBMS reaches its maximum number of connections (~300). I think this is also related to Hot Reloading and compiling the end points each time they are called. If I stop the dev server and restart, then all the connections are apparently destroyed because I can connect again. In my application's `connection-pool.js` I create a top level variable as null, and either use that variable or assign it the connection pool if null (singleton). However it seems when the file is imported in the various API routes, a new instance of `connection-pool.js` is created instead of reusing the one that was imported elsewhere, or maybe it's just related to compiling the "page" each time and leaving the old connection pools hanging out in memory abyss.

If there was some hook I could use to clean up my connection pool before this happens then I wouldn't reach my DB's max connection limit. Open to all ideas.

> event - build page: /api/order/listForUser
wait  - compiling...
event - compiled successfully
Connection pool was falsey. Created a new connection pool.
Released connection for reuse.
event - build page: /order/[orderId]
wait  - compiling...
event - compiled successfully
event - build page: /api/order/details
wait  - compiling...
event - compiled successfully
Connection pool was falsey. Created a new connection pool.
Released connection for reuse.
Released connection for reuse.

**Comment by fultonm** [comment, marked as answer]
Okay, we should assign variables we wish to be global to the global object in node. So in my case I can use 'global.connectionPool'. This way it persists through reloads.

**Reply by aprilmintacpineda** [reply]
I wonder how we'd handle this today with typescript.
